pyface/data_view/i_data_view_widget.py

```diff
@@ -116,7 +222,7 @@ def _set_control_selection_mode(self, selection_mode):
 
     def _selection_updated(self, event):
         """ Observer for selection trait. """
-        if self.control is not None and not self._selection_updating_flag:
```

Comment:
I removed this to test if it is necessary at all. I tried to test but I could not see why this is needed.